vaadin-side-nav-flow-parent/vaadin-side-nav-flow-integration-tests/src/test/java/com/vaadin/flow/component/sidenav/tests/SideNavIT.java

```diff
@@ -115,6 +118,39 @@ public void collapseSideNav_expandedStateSynchronized() {
         assertExpandedStateOnServer("print-side-nav-expanded-state", "false");
     }
 
+    @Test
+    public void pageOpened_itemWithMatchingPathIsCurrent() {
+        Assert.assertTrue(currentItem.isCurrent());
+    }
+
+    @Test
+    public void navigateWithParametersInUrl_itemWithMatchingPathIsCurrent() {
+        getDriver().navigate().to(getDriver().getCurrentUrl() + "?key=value");
+        waitUntil(driver -> $(SideNavElement.class).exists(), 1);
+
+        Assert.assertTrue(
+                $(SideNavItemElement.class).id("current-item").isCurrent());
+    }
+
+    @Test
+    public void addParametersToCurrentItem_itemWithMatchingPathIsCurrent() {
+        $(NativeButtonElement.class).id("add-parameters-to-current-item")
+                .click();
+
+        Assert.assertTrue(currentItem.isCurrent());
+    }
+
+    @Test
+    public void itemWithParameters_removeParametersFromItem_itemWithMatchingPathIsCurrent() {
+        $(NativeButtonElement.class).id("add-parameters-to-current-item")
+                .click();
+
+        $(NativeButtonElement.class).id("remove-parameters-from-current-item")
+                .click();
+
+        Assert.assertTrue(currentItem.isCurrent());
+    }
```

Comment:
Not sure about those, seems these test logic from the web component?

Reply:
You are right. The unit tests for the query parameters should be enough here.

Reply:
Removed.